src/TestUtils/CloudFunctionDeploymentTrait.php

```diff
@@ -101,6 +89,12 @@ private static function doDelete()
      */
     public function setUpClient()
     {
+        // CloudEvent functions do not have URLs.
+        // Guzzle client creation stalls without a valid URL.
```

Comment:
Guzzle stalls without a URL, or `getBaseUri` stalls? I don't think Guzzle needs a URL, but this seems in any case.

Reply:
When getBaseUrl() is called on a CloudEvent function, it silently fails.
Then something in the client setup (guessing the token middleware factory) has a really long timeout. I'll look at filing an issue on that.

Your request for a `$functionSignatureType` approach will lead to a few ripples, this is something I expect to change a little.